src/Kiota.Builder/Writers/CSharp/CSharpConventionService.cs

```diff
@@ -90,19 +90,24 @@ public override string GetTypeString(CodeTypeBase code, CodeElement targetElemen
         private string TranslateTypeAndAvoidUsingNamespaceSegmentNames(CodeType currentType, CodeElement targetElement)
         {
             var typeName = TranslateType(currentType);
-            if(currentType.TypeDefinition != null &&
-                GetNamesInUseByNamespaceSegments(targetElement).Contains(typeName))
+            if (currentType.TypeDefinition != null && GetNamesInUseByNamespaceSegments(targetElement).Contains(typeName) && !DoesTypeExistsInSameNamesSpaceAsTarget(currentType,targetElement))
```

Comment:
There exists the namespace `MicrosoftGraphSdk.App.Calls.Item.Record.RecordResponse` and the type `RecordResponse` in the `MicrosoftGraphSdk.App.Calls.Item.Record` namespace. 
Fully qualifying the type name causes ambiguity as the compiler can't infer between the type `MicrosoftGraphSdk.App.Calls.Item.Record.RecordResponse` and the namespace `MicrosoftGraphSdk.App.Calls.Item.Record.RecordResponse`.

This fixes this to ensure we only use the fully qualified name if it only exists in a namespace that is not the same as that of the target.